### PR TITLE
MM-17468: Accept a number of days instead of two date arguments in `/incident test bulk-data`

### PR DESCRIPTION
**Pull request by alankan886**
#### Summary
This PR is for simplifying two date arguments in `/incident test bulk-data` into one number argument. Instead of `/incident test bulk-data 10 3 2020-01-31 2020-11-22 2`, now it's `/incident test bulk-data 10 3 342 2`. And the number of days is expected to be larger than 0.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/17468

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Thank you for your contribution, @alankan886 :tada: We will review it as soon as we can!

File: server/command/command.go [hunk 1]
```diff
@@ -1134,23 +1134,34 @@ func (r *Runner) actionTestData(params []string) {
...
+	if days < 1 {
+		r.postCommandResponse(fmt.Sprintf("The provided value for days, '%d', is not greater than 0.", days))
 		return
 	}
 
-	end, err := time.ParseInLocation("2006-01-02", params[3], time.Now().Location())
+	begin, err := time.ParseInLocation("2006-01-02", time.Now().AddDate(0, 0, -days).Format("2006-01-02"), time.Now().Location())
 	if err != nil {
-		r.postCommandResponse(fmt.Sprintf("The provided value for the last possible date, '%s', is not a valid date. It needs to be in the format 2020-01-31.", params[3]))
+		r.warnUserAndLogErrorf("unable to parse in location on time.Now(): %v", err)
```
Comment (cpoile) [inline server/command/command.go:1150]:
```suggestion
		r.warnUserAndLogErrorf("unable to parse in location on time.Now().AddDate: %v", err)
```

**Review by cpoile — APPROVED** [review]
Looks good to me from a code standpoint, thanks @alankan886 !

**Comment by alankan886** [comment]
Thank you for the comment @cpoile! The suggested change is made and pushed.

File: server/command/command.go [hunk 2]
```diff
@@ -130,7 +130,7 @@ func getAutocompleteData(addTestCommands bool) *model.AutocompleteData {
 		testCreate.AddTextArgument("Name of the incident", "Incident name", "")
 		test.AddCommand(testCreate)
 
-		testData := model.NewAutocompleteData("bulk-data", "[ongoing] [ended] [begin] [end] [seed]", "Generate random test data in bulk")
+		testData := model.NewAutocompleteData("bulk-data", "[ongoing] [ended] [days] [seed]", "Generate random test data in bulk")
 		testData.AddTextArgument("An integer indicating how many ongoing incidents will be generated.", "Number of ongoing incidents", "")
 		testData.AddTextArgument("An integer indicating how many ended incidents will be generated.", "Number of ended incidents", "")
 		testData.AddTextArgument("Date in format 2020-01-31", "First possible creation date", "")
```
Comment (agarciamontoro) [inline server/command/command.go:136]: We also need to update these descriptions. We need to modify this one as shown here and remove the next line

```suggestion
		testData.AddTextArgument("An integer n. The incidents generated will have a start date between n days ago and today.", "Range of days for the incident start date", "")
```

@justinegeffen may have a better wording for this.

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thank you, @alankan886, it looks great! :tada:

I left some suggestions below.

**Comment by agarciamontoro** [comment]
@itao, I think it is ok getting rid of the old way of creating the test incidents and use instead just a number of days. What this means effectively is that it will be much easier to specify the number of days you want the incidents to go back (you just write that number), but also that it is now impossible to specify an end date, so the range will always be `[n days ago, today]`, with the range always ending in the day the test command is executed.

Is that ok? Or do we want to maintain the old way of generating this data?

File: server/command/command.go [hunk 3]
```diff
@@ -1134,23 +1134,34 @@ func (r *Runner) actionTestData(params []string) {
...
+		return
+	}
+
+	if days < 1 {
+		r.postCommandResponse(fmt.Sprintf("The provided value for days, '%d', is not greater than 0.", days))
 		return
 	}
 
-	end, err := time.ParseInLocation("2006-01-02", params[3], time.Now().Location())
+	begin, err := time.ParseInLocation("2006-01-02", time.Now().AddDate(0, 0, -days).Format("2006-01-02"), time.Now().Location())
```
[the remark this answers is not in the document]
Comment (alankan886) [reply server/command/command.go:1148]: Hello, not sure if this matters, but they are slightly different where `time.ParseInLocation()` would give `2021-05-12 00:00:00` (no specific time) and `time.Now()` would give `2021-05-12 19:53:00.596493` (specific time). If you don't mind the differences, I'm happy to change them. Let me know if you have any thoughts on this!

[the remark this answers is not in the document]
Comment (agarciamontoro) [reply server/command/command.go:1148]: We do not need a high specificity here, as the dates will be picked randomly in the range of dates. So I'd vote for making the code more readable using just `time.Now()`.

**Review by itao — APPROVED** [review]
This is awesome! It makes the command one-clickable in a playbook which is helpful for the customer demo playbook. Nice!

**Review by agarciamontoro — APPROVED** [review]
This is great, @alankan886, thank you very much! Skipping QA as it is an internal, testing command.

**Pull request merged**